creator/data_templates/mutations/data_template.py

```diff
@@ -0,0 +1,217 @@
+
+    class Arguments:
+        input = CreateDataTemplateInput(
+            required=True, description="Attributes for the new data_template"
+        )
+
+    data_template = graphene.Field(DataTemplateNode)
+
+    def mutate(self, info, input):
+        """
+        Creates a new data_template.
+        """
+        user = info.context.user
+        if not user.has_perm("data_templates.add_datatemplate"):
+            raise GraphQLError("Not allowed")
+
+        # Check if organization exists
+        org_id = input.pop("organization")
+        model, org_id = from_global_id(org_id)
+
+        try:
+            org = Organization.objects.get(pk=org_id)
+        except Organization.DoesNotExist:
+            raise GraphQLError(f"Organization {org_id} does not exist")
+
+        # User may only add templates to an org they are a member of
+        if not user.organizations.filter(pk=org.pk).exists():
+            raise GraphQLError(
+                "Not allowed - may only add templates to an organization the "
+                "user is a member of."
+            )
+
+        # Create data template
+        data_template = DataTemplate(**{k: input[k] for k in input})
+        data_template.organization = org
+        data_template.creator = user
+        data_template.save()
+
+        return CreateDataTemplateMutation(data_template=data_template)
+
+
+class UpdateDataTemplateMutation(graphene.Mutation):
+    """Update an existing data_template"""
+
+    class Arguments:
+        id = graphene.ID(
+            required=True, description="The ID of the data_template to update"
+        )
+        input = UpdateDataTemplateInput(
+            required=True, description="Attributes for the data_template"
+        )
+
+    data_template = graphene.Field(DataTemplateNode)
+
+    def mutate(self, info, id, input):
+        """
+        Updates an existing data_template
+        """
+        user = info.context.user
+        if not user.has_perm("data_templates.change_datatemplate"):
+            raise GraphQLError("Not allowed")
+
+        model, dt_id = from_global_id(id)
+
+        try:
+            data_template = DataTemplate.objects.get(id=dt_id)
+        except DataTemplate.DoesNotExist:
+            raise GraphQLError(f"DataTemplate {dt_id} does not exist")
+
+        # User may only change templates for an org they are a member of
+        if not (
+            user.organizations.filter(
+                pk=data_template.organization.pk
+            ).exists()
+        ):
+            raise GraphQLError(
+                "Not allowed - may only update templates owned by an "
+                "organization that the user is a member of"
+            )
+
+        # User may only update a template if it is not being used in
+        # any studies
+        if data_template.released:
+            raise GraphQLError(
+                "Not allowed - may only update templates that are not being "
+                "used by any studies"
+            )
```

Comment:
should this be limited to certain fields? probably not a big deal to edit the description after it has been used

Reply:
Yea I really wasn't sure about this..The description of the template is supposed to describe a data type or the semantic meaning of the template. So I guess after the template is in use, I figured you wouldn't want to change this. I could also see typos making it into a template description and then the template being released. So idk, I could go either way on this

Reply:
i think allowing the description only to be modified post-release is reasonable

Reply:
Ok, descriptions of the `DataTemplate` and the `TemplateVersion` are allowed to be modified even if they are being used in studies